Public/Sdk/Public/Prelude/Prelude.Configuration.Resolvers.dsc

```diff
@@ -160,8 +163,11 @@ interface MsBuildResolver extends ResolverBase, UntrackingSettings {
      * Note: if this field is not specified any change in an environment variable will potentially cause
      * cache misses for all pips. This is because there is no way to know which variables were actually used during the build.
      * Therefore, it is recommended to specify the environment explicitly.
+     * The value can be either a string or a PassthroughEnvironmentVariable, the latter representing that the associated variable will be exposed
+     * but its value won't be considered part of the build inputs for tracking purposes. This means that any change in the value of the 
+     * variable won't cause a rebuild.
      */
-    environment?: Map<string, string>;
+    environment?: Map<string, (PassthroughEnvironmentVariable | string)>;

```

Comment:
>Map [](start = 18, length = 3)

Are you sure you wnat ot use a map, and not just a list of key-value pairs? That would be a lot nicer to author since we currently lack a nice literal construction for maps like C# has...

Reply:
or altenatively a keyed object... (forgot syntax...)

---
In reply to: [293480516](https://github.com/microsoft/BuildXL/pull/462#discussion_r293480516) [](ancestors = 293480516)

Reply:
Agree with the lack of nice literal construction. On the other hand, a map fits more naturally since it never makes sense to specify a variable twice. And we can always consider adding literal-specific stuff for maps. We already diverged from TS :).
So unless you feel strong about this one, I'd prefer to leave it as is.